packages/integration-karma/test/light-dom/multiple-templates/index.spec.js

```diff
@@ -0,0 +1,33 @@
+import { createElement, setFeatureFlagForTest } from 'lwc';
+
+import Multi from 'x/multi';
+
+describe('multiple templates', () => {
+    beforeEach(() => {
+        setFeatureFlagForTest('ENABLE_LIGHT_DOM_COMPONENTS', true);
+    });
+
+    afterEach(() => {
+        setFeatureFlagForTest('ENABLE_LIGHT_DOM_COMPONENTS', false);
+    });
+
+    it('can render multiple templates with different styles', () => {
+        const element = createElement('x-multi', { is: Multi });
+
+        document.body.appendChild(element);
+
+        expect(element.querySelector('div').textContent).toEqual('a');
+        expect(getComputedStyle(element.querySelector('div')).color).toEqual('rgb(233, 150, 122)');
+        expect(getComputedStyle(element.querySelector('div')).backgroundColor).toEqual(
+            'rgba(0, 0, 0, 0)'
+        );
+        element.next();
+        return Promise.resolve().then(() => {
+            expect(element.querySelector('div').textContent).toEqual('b');
+            expect(getComputedStyle(element.querySelector('div')).color).toEqual('rgb(0, 0, 0)');
+            expect(getComputedStyle(element.querySelector('div')).backgroundColor).toEqual(
+                'rgb(255, 160, 122)'
+            );
```

Comment:
These tests actually caught an important edge case I hadn't considered: when rendering multiple templates, when a template is removed from the DOM, its `<style>`s are removed too. I believe this matches the developer intuition that global styles associated with a template should be removed when that template is un-rendered.

We would have to consider this in the future, if/when we implement global light DOM styles such that they render to the `<head>` rather than _in situ_.